Sources/TuistGenerator/Generator/ConfigGenerator.swift

```diff
@@ -280,46 +287,62 @@ final class ConfigGenerator: ConfigGenerating {
         return settings
     }
 
-    private func deploymentTargetDerivedSettings(target: Target) -> SettingsDictionary {
-        guard let deploymentTarget = target.deploymentTarget else {
-            return [:]
-        }
-
+    private func destinationsDerivedSettings(target: Target) -> SettingsDictionary {
         var settings: SettingsDictionary = [:]
 
-        switch deploymentTarget {
-        case let .iOS(version, devices, supportsMacDesignedForIOS):
-            var deviceFamilyValues: [Int] = []
-            if devices.contains(.iphone) { deviceFamilyValues.append(1) }
-            if devices.contains(.ipad) { deviceFamilyValues.append(2) }
-
-            settings["TARGETED_DEVICE_FAMILY"] = .string(deviceFamilyValues.map { "\($0)" }.joined(separator: ","))
-            settings["IPHONEOS_DEPLOYMENT_TARGET"] = .string(version)
-            settings["SUPPORTS_MAC_DESIGNED_FOR_IPHONE_IPAD"] = supportsMacDesignedForIOS ? "YES" : "NO"
-
-            if devices.contains(.ipad), devices.contains(.mac) {
-                settings["SUPPORTS_MACCATALYST"] = "YES"
-                settings["DERIVE_MACCATALYST_PRODUCT_BUNDLE_IDENTIFIER"] = "YES"
-            } else {
-                // Unless explicitly specified, when the platform the Product is a framework, these default to YES.
-                settings["SUPPORTS_MACCATALYST"] = "NO"
-            }
-        case let .macOS(version):
-            settings["MACOSX_DEPLOYMENT_TARGET"] = .string(version)
-        case let .watchOS(version):
-            settings["WATCHOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .tvOS(version):
-            settings["TVOS_DEPLOYMENT_TARGET"] = .string(version)
-        case let .visionOS(version):
-            let deviceFamilyValues = [1, 2, 7]
-
+        var deviceFamilyValues: [Int] = []
+        if target.destinations.contains(.iPhone) { deviceFamilyValues.append(1) }
+        if target.destinations.contains(.iPad) { deviceFamilyValues.append(2) }
+        if target.destinations.contains(.appleTv) { deviceFamilyValues.append(3) }
+        if target.destinations.contains(.appleWatch) { deviceFamilyValues.append(4) }
+        if target.destinations.contains(.appleVision) { deviceFamilyValues.append(7) }
```

Comment:
What about extracting `1`, `2`, `3`, `4`, and `7` into an enum to easily recognise the mapping between a destination and the value in `TARGETED_DEVICE_FAMILY`.